reroll-initiative/reroll-initiative.js

```diff
@@ -0,0 +1,205 @@
+  * Set module constants
+  * --------------------
+  */
+ const RRI_DEFAULT_SETTINGS = {
+    reroll: true,
+    actorTypes: "all"
+}
+  
+
+/**
+ * @class RerollInitiative
+ * @description Hooks on combat update and rerolls initiative for all combatants
+ * @todo Add configurability for when to reroll and for whom, make enable setting define whether the hook is registered or not
+ */
+class RerollInitiative {
+    settings = {};
+
+    constructor() {
+        this.postUpdateCombatHook();
+        //this.settings = {};
+        this._registerSettings();
+    }
+
+    /**
+     * module settings
+     * @todo need to store settings as an object so they can easily be retrieved
+     */
+    _registerSettings () {
+        game.settings.register("reroll-initiative", "rriSettings", {
+            name: "Reroll-Initiative Settings",
+            hint: "Settings for Reroll-Initiative module",
+            default: RRI_DEFAULT_SETTINGS,
+            type: Object,
+            scope: "world",
+            onChange: setting => {
+                console.log("settings changed, new values: ",setting)
+                //this.settings = JSON.stringify(setting);
+            }
+        });
+
+        this._loadSettings();
+
+        /* add settings as object instead
+        game.settings.register('reroll-initiative', "rriStatus", {
+            name: "Reroll-Initiative Status",
+            hint: "Enable the rerolling initiative true/false",
+            default: RRI_CONFIG.reroll,
+            //default: "",
+            type: Boolean,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+
+        game.settings.register('reroll-initiative', "actorTypesToReroll", {
+            name: "Actor Types to Reroll",
+            hint: "Specify the actor types to reroll PCs/NPCs/All",
+            default: RRI_CONFIG.actorTypes,
+            //default: "",
+            type: String,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+        */
+    }
+
+    /**
+     * Returns the default module settings
+     */
+    _defaultSettings() {
+        this.settings = RRI_SETTINGS;
+        console.log("Restting reroll-initiative settings to defaults:",RRI_DEFAULT_SETTINGS);
+    }
+
+    /**
+     * Saves current class instance settings back to game settings
+     */
+    _saveSettings () {
+        game.settings.set("reroll-initiative","rriSettings",this.settings);
+    }
+
+    /**
+     * Loads current class instance settings from game settings
+     */
+    _loadSettings (){
+        let settings = game.settings.get("reroll-initiative","rriSettings");
+        this.settings = settings;
+        console.log(this.settings);
+    }
+    
+    /**
+     * Get the current class instance settings (for external use)
+     */
+    get settings() {
+        this._loadSettings();
```

Comment:
Get methods by practice are not suppose to mutate the code. Should load be called in the constructor?